### PR TITLE
zabbix: add server frontend

### PR DESCRIPTION
**Pull request by Rixerx**
Added zabbix-server-frontend package, which allow to use website interface. 

To make frontend working, it is required to add some options to uhttpd config witch allow php working.
```
uci add_list uhttpd.main.index_page='index.html, default.html, index.php'
uci add_list uhttpd.main.interpreter='.php=/usr/bin/php-cgi'
uci set uhttpd.main.index_file='index.html, default.html, index.php'
uci commit
```
To make it working it is required to merge: https://github.com/openwrt/packages/pull/6928 (merged)

This configuration was working for about 2 weeks without problem.

Compile tested: Yes, brcm2708
Run tested: Yes, brcm2708

Maintainer: @champtar

Signed-off-by: Krystian Kozak <krystian.kozak20@gmail.com>

### CODE REVIEW

File: admin/zabbix/Makefile [hunk 1]
```diff
@@ -29,7 +31,7 @@ define Package/zabbix/Default
   SECTION:=admin
   CATEGORY:=Administration
   TITLE:=Zabbix
-  URL:=http://www.zabbix.com/
+  URL:=https://www.zabbix.com/
```
Comment (karlp) [inline admin/zabbix/Makefile]: all the changes above are independet and should just go in separately.

File: admin/zabbix/Makefile [hunk 2]
```diff
@@ -72,13 +74,21 @@ endef
 define Package/zabbix-server
   $(call Package/zabbix/Default)
   TITLE+= server
-  DEPENDS += +pgsql-cli +libevent2
+  DEPENDS += +mysql-server +libmariadb +libevent2
```
Comment (karlp) [inline admin/zabbix/Makefile]: this change seems completely out of scope of "add front end support"

File: admin/zabbix/Makefile [hunk 3]
```diff
@@ -72,13 +74,21 @@ endef
...
+  DEPENDS += +zabbix-server +php7 +php7-cgi +php7-fpm +php7-mod-mysqli \
+  +php7-mod-gd +php7-mod-bcmath +php7-mod-ctype +php7-mod-xmlreader +php7-mod-xmlwriter \
+  +php7-mod-session +php7-mod-sockets +php7-mod-mbstring +php7-mod-gettext
 endef
 
 define Package/zabbix-proxy
   $(call Package/zabbix/Default)
   TITLE+= proxy
-  DEPENDS += +pgsql-cli
+  DEPENDS += +mysql-server +libmariadb
```
Comment (karlp) [inline admin/zabbix/Makefile]: again, this seems out of scope.  And are you sure you need teh _server_ to run the proxy?  (note that pgsql-cli doesn't pull in a pgsql server)

File: admin/zabbix/Makefile (hunk 1 above)
Comment (Rixerx) [reply admin/zabbix/Makefile]: Done

File: admin/zabbix/Makefile (hunk 2 above)
Comment (Rixerx) [reply admin/zabbix/Makefile]: Done, now it makes complex of backend and database, where before it was only
CLI to connect exterior database. Now it looks like debian zabbix-server
package.

File: admin/zabbix/Makefile (hunk 3 above)
Comment (Rixerx) [reply admin/zabbix/Makefile]: Proxy it is a buffer with local database which sends data to the server.
It also gets dependency to local database.
[1].https://www.zabbix.com/documentation/3.4/manual/concepts/proxy

File: admin/zabbix/Makefile [hunk 4]
```diff
@@ -222,16 +230,30 @@ define Package/zabbix-server/install
...
+
 define Package/zabbix-proxy/install
 	$(call Package/zabbix/install/sbin,$(1),proxy)
 	$(call Package/zabbix/install/etc,$(1),proxy)
 endef
 
+define Package/zabbix-server-frontend/postinst
+#!/bin/sh
+[ -n "$${IPKG_INSTROOT}" ] || {
+	ln -sf /www/zabbix/ /www/zabbix/
```
Comment (karlp) [inline admin/zabbix/Makefile]: what is this for?

**Comment by karlp** [comment]
Please don't add unrelated changes to this PR, version bumps and adding database backends are all for their own PRs

File: admin/zabbix/Makefile (hunk 4 above)
Comment (karlp) [reply admin/zabbix/Makefile]: Yes, but you're using the same path twice.  are you _sure_ this is correct.  It's not from the listed reference either.

**Comment by Rixerx** [comment]
> Please don't add unrelated changes to this PR, version bumps and adding database backends are all for their own PRs

Done,
#7071 
#7070

File: admin/zabbix/Makefile (hunk 4 above)
Comment (Rixerx) [reply admin/zabbix/Makefile]: This solution worked before. I corrected it to stop mislead people. Now it should be okay. I have tested it and it works perfect after adding php interpreter to uhttpd.

File: admin/zabbix/Makefile [hunk 5]
```diff
@@ -75,6 +75,14 @@ define Package/zabbix-server
   DEPENDS += +pgsql-cli +libevent2
 endef
 
+define Package/zabbix-server-frontend
+  $(call Package/zabbix/Default)
+  TITLE+= server-frontend
+  DEPENDS += +zabbix-server +php7 +php7-cgi +php7-fpm +php7-mod-mysqli \
```
Comment (champtar) [inline admin/zabbix/Makefile]: you are using `php7-cgi` here, so I don't think you need `php7-fpm`
not sure you need `php7` (cli ?)
`zabbix-server` is not a strict requirement
`php7-mod-mysqli` should be selected by db backend (if we build with postgresql support, select the correct backend)

File: admin/zabbix/Makefile [hunk 6]
```diff
@@ -222,16 +230,30 @@ define Package/zabbix-server/install
...
 define Package/zabbix-proxy/install
 	$(call Package/zabbix/install/sbin,$(1),proxy)
 	$(call Package/zabbix/install/etc,$(1),proxy)
 endef
 
+define Package/zabbix-server-frontend/postinst
+#!/bin/sh
+[ -n "$${IPKG_INSTROOT}" ] || {
+	ln -sf /www/ /www/zabbix/
+}
```
Comment (champtar) [inline admin/zabbix/Makefile]: this symlink, if really needed, should be part of the package files and not a postinst script
```
# ls -lia /www/zabbix/www
171 lrwxrwxrwx    1 root     root             5 Oct  1 04:47 /www/zabbix/www -> /www/
```

**Comment by champtar** [comment]
@Rixerx i've merged both #7070 & #7071

File: admin/zabbix/Makefile (hunk 5 above)
Comment (Rixerx) [reply admin/zabbix/Makefile]: Done

File: admin/zabbix/Makefile (hunk 6 above)
Comment (Rixerx) [reply admin/zabbix/Makefile]: Tested, it is not required

**Comment by Rixerx** [comment]
@champtar Corrected

**Comment by champtar** [comment]
Hi @Rixerx 
Can you rebase on latest master so it merge properly ?

**Comment by Rixerx** [comment]
@champtar  Sorry for reopening the PR. I were not sure how to rebase branch with master.
